lambdaclass/madara_explorer
# Make the fetcher receive a range

## Changes:
- Since we have a considerable amount of blocks related modules,
   I've moved them under starknet_explorer/blocks.
- There are now 2 process to store blocks:
   1. BlockListener.
   2. BlockFetcher.
- BlockListener is the old BlockFetcher, that is, waits for blocks to appear and stores them.
   Now enabled with the ENABLE_LISTENER env variable.
- BlockFetcher takes a range of numbers and stores every block on said range.
## How to test:
1. Make sure your db is empty: `mix ecto.reset`
2. Set your RPC provider:

```
export RPC_API_HOST=your_rpc_hostname
export TESTNET_RPC_API_HOST=testnet_rpc_hostname
export TESTNET_2_RPC_API_HOST=testnet_2_rpc_hostname
```

You can also use Juno instead, and set every env var to "http://localhost:6060",
since this is only a test.
3. `make setup run`
4. Run the BlockFetcher, for example:  `StarknetExplorer.BlockFetcher.fetch_in_range(%{start: 100, finish: 1})`
5. Open the db:
```bash
sqlite3 /priv/repo/explorer_dev.db
```
6. Check you only have the requested amount of blocks:
`select count(*) from blocks;`
